src/components/util/use-media-query.ts

```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect } from 'react';
+
+/**
+ *
+ * via https://www.netlify.com/blog/2020/12/05/building-a-custom-react-media-query-hook-for-more-responsive-apps/
+ */
+export function useMediaQuery(query) {
```

Comment:
Checking the media query with Javascript has one big downside: The initial render will always be either mobile or desktop. If it's desktop and the user opens the page on mobile, it will flicker.

In other projects we are actively removing the media query Javascript hook and replacing it with pure CSS to improve the UX. A standard way to do this, is to add both the desktop and the mobile HTML to the DOM tree, and hide either one or the other with a media query.

It makes the DOM tree lager but in most cases this downside is worth the upside of having a perfect page on the initial render.

Reply:
Thanks for the hint. Makes sense and I actually prefer the CSS only solution. I got dragged into that direction by forcing myself to think "more react-like" whatever this means but it caused me to use JavaScript rather than CSS. I will remove it already again 👌